### PR TITLE
Add Pipeline Variables / Set Variable Activity / Append Variable Activity

### PR DESCRIPTION
**Pull request by leonliang33**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Review by hvermis — APPROVED** [review]

**Comment by annatisch** [comment]
Thanks @leonliang33 - these two swagger specs have a large volume of model validation errors - are you able to address some of these?

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/datafactory.json [hunk 1]
```diff
@@ -3101,6 +3101,39 @@
         "message"
       ]
     },
+    "VariableDefinitionSpecification": {
+      "description": "Definition of all variables for an Pipeline.",
```
Comment (annatisch) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/datafactory.json]: "a Pipeline"

Additionally - this description says "Definition of *all* variables - yet it only references a single Specification.
Should it be referencing an array or `VariableSpecifications`? If not - maybe alter this description.

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/datafactory.json [hunk 2]
```diff
@@ -3101,6 +3101,39 @@
...
+      "type": "object",
+      "additionalProperties": {
+        "$ref": "#/definitions/VariableSpecification"
+      }
+    },
+    "VariableSpecification": {
+      "description": "Definition of a single variable for an pipeline.",
```
Comment (annatisch) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/datafactory.json]: "a Pipeline"

**Review by annatisch — COMMENTED** [review]
LGTM - just a few minor docs comments.

**Comment by leonliang33** [comment]
Need more changes @haricr requests

**Comment by leonliang33** [comment]
Reopened, decided no more changes are necessary

**Comment by hvermis** [comment]
@annatisch I will fix the validation errors later. We need to update our capture examples code to get 304-s and most of the errors are about that.

**Review by annatisch — APPROVED** [review]

**Pull request merged**